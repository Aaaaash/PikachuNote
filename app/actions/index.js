import {
  INSERT_NOTE
} from '../common/constants';

export function insertNote(id, payload) {
  return {
    type: INSERT_NOTE,
    id,
    payload,
  };
}
