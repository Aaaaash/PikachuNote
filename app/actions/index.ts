import {
  INSERT_NOTE
} from '../common/constants';

export function insertNote(id: any, payload: any) {
  return {
    type: INSERT_NOTE,
    id,
    payload,
  };
}
