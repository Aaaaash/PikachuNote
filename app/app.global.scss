@import "~font-awesome/css/font-awesome.css";
@import '~@blueprintjs/core/dist/blueprint.css';

body {
  position: relative;
  color: #232c39;
  height: 100vh;
  padding: 0;
  margin: 0;
  background-color: #f5f8fa;
  font-family: Arial, Helvetica, Helvetica Neue, serif;
  overflow-y: hidden;
  user-select: none;
}

h2 {
  margin: 0;
  font-size: 2.25rem;
  font-weight: bold;
  letter-spacing: -0.025em;
  color: #fff;
}

li {
  list-style: none;
}

a {
  color: white;
  opacity: 0.75;
  text-decoration: none;
}

a:hover {
  opacity: 1;
  text-decoration: none;
  cursor: pointer;
}

button, input {
  outline: none;
}

.pt-icon-standard {
  padding-right: 5px;
  color: #434A54 !important;
}

.pt-input {
  -webkit-app-region: no-drag;
  cursor: text !important;
}