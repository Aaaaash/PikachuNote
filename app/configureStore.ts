import { routerMiddleware }  from 'react-router-redux';
import { History } from 'history';
import {
  applyMiddleware,
  GenericStoreEnhancer,
  compose,
  createStore,
  Middleware,
} from 'redux';

import createReducers from './reducers';

export default (initialState: Object | void, history: History) => {
  const middlewares: Middleware[] = [
    routerMiddleware(history),
  ];

  const enhaners: GenericStoreEnhancer[] = [
    applyMiddleware(...middlewares),
  ];

  const composeEnhancers: Function =
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
      shouldHotReload: false,
    })
    : compose;

  const store = createStore(
    createReducers(),
    initialState,
    composeEnhancers(...enhaners)
  );

  return store;
};
