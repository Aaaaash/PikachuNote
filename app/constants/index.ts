export const INSERT_NOTE = 'app/sidebar/INSERT_NOTE';

export const FETCH_ALL_DIRS = 'app/sidebar/FETCH_ALL_DIRS';
export const FETCH_ALL_DIRS_COMPLETE = 'app/sidebar/FETCH_ALL_DIRS_COMPLETE';

export const FETCH_DIRECTORY_NOTES = 'app/sidebar/FETCH_DIRECTORY_NOTES';
export const FETCH_DIRECTORY_NOTES_COMPLETE = 'app/sidebar/FETCH_DIRECTORY_NOTES_COMPLETE';

export const SET_CURRENT_DIRECTORY = 'app/sidebar/SET_CURRENT_DIRECTORY';

export const SET_ACTIVE_ITEM = 'app/dirdetails/SET_ACTIVE_ITEM';
