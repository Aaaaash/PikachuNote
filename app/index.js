import React from 'react';
import { render } from 'react-dom';
import { AppContainer } from 'react-hot-loader';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'react-router-redux';
import Routes from './routes';
import { configureStore, history } from './store/configureStore';
import './app.global.css';

const store = configureStore();

render(
  <AppContainer>
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Routes />
      </ConnectedRouter>
    </Provider>
  </AppContainer>,
  document.getElementById('root')
);

if (module.hot) {
  // module.hot.accept('./containers/Root', () => {
  //   const NextRoot = require('./containers/Root'); // eslint-disable-line global-require
  //   render(
  //     <AppContainer>
  //       <NextRoot store={store} history={history} />
  //     </AppContainer>,
  //     document.getElementById('root')
  //   );
  // });
}
