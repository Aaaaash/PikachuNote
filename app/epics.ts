import { ActionsObservable } from 'redux-observable';
import { Observable } from 'rxjs';
import 'rxjs';

import { fetchAllDataByStoreName } from './utils/indexedDB';
import { fetchAllDataComplete } from './actions';
import { Directory, ElectronAction } from './common/constants';
import { FETCH_ALL_DIRS } from './constants';

const fetchDirectory = (action$: ActionsObservable<ElectronAction>) => {
  return action$.ofType(FETCH_ALL_DIRS)
    .mergeMap((action: ElectronAction) =>
      Observable.fromPromise(fetchAllDataByStoreName(action.dbName, action.storeName))
      .map((response: Directory[]) => fetchAllDataComplete(response))
    );
}

export {
  fetchDirectory,
};
