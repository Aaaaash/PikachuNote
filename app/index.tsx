import * as React from "react";
import { render } from "react-dom";
import { AppContainer } from "react-hot-loader";
import { Provider } from "ractor-react";
import moment from 'moment';

import { system } from "./system/appSystem";
import Route from "./routes";
import "./app.global.scss";

import { isDataBasebeCreated } from "./api/indexdb";
import { INDEXED_DATABASE_NAME } from "./common/constants";
import PikachuStore from "./store/sidebar.store";
import RepeatStore from "./store/repeat.store";
import initialDatabase from "./initialDatabase";

moment.locale('zh-cn');

isDataBasebeCreated(INDEXED_DATABASE_NAME)
  .then((haveDb): any => {
    if (!haveDb) {
      return initialDatabase();
    }
    return null;
  })
  .then(() => {
    render(
      <AppContainer>
        <Provider system={system} stores={[PikachuStore, RepeatStore]}>
          <Route />
        </Provider>
      </AppContainer>,
      document.getElementById("root")
    );
  });
