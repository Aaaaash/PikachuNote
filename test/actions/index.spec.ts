import * as actions from '../../app/actions';

describe('actions', () => {
  it('should fetchAllData should create fetchAllData action', () => {
    expect(actions.fetchAllData('dbName', 'storeName')).toMatchSnapshot();
  });

  it('should fetchAllDataComplete should create fetchAllDataComplete action', () => {
    expect(actions.fetchAllDataComplete([])).toMatchSnapshot();
  });
});
