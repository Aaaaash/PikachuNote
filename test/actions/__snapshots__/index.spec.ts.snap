// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should fetchAllData should create fetchAllData action 1`] = `
Object {
  "dbName": "dbName",
  "storeName": "storeName",
  "type": "app/sidebar/FETCH_ALL_DIRS",
}
`;

exports[`actions should fetchAllDataComplete should create fetchAllDataComplete action 1`] = `
Object {
  "data": Array [],
  "type": "app/sidebar/FETCH_ALL_DIRS_COMPLETE",
}
`;
